.buildkite/scripts/index-micro-benchmark-results.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+METRICS_HOST=$(vault read -field=es_host /secret/performance/employees/cloud/esbench-metrics)
+METRICS_INDEX_NAME="dummy-micro-benchmarks"
+METRICS_USERNAME=$(vault read -field=es_username /secret/performance/employees/cloud/esbench-metrics)
+METRICS_PASSWORD=$(vault read -field=es_password /secret/performance/employees/cloud/esbench-metrics)
+
+jq -c '.[]' "benchmarks/build/result.json" | while read -r doc; do
+  curl -s -X POST "https://$METRICS_HOST/$METRICS_INDEX_NAME/_doc" \
+    -u "$METRICS_USERNAME:$METRICS_PASSWORD" \
```

Comment:
Do we leak the credentials here to console log? @brianseeders

Reply:
It doesn't look like it, but the `vault read` commands above need to happen inside of `.buildkite/hooks/pre-command`. If they happen there, then they will get automatically redacted if they are accidentally printed.

Take a look at the other `vault` commands in that script, there's a few different ways to do it. I usually prefer introducing an env var like `USE_ES_BENCH_CREDENTIALS`. You can also do it based on something like the step label `periodic-micro-benchmarks`, but then it will break if you rename the step.